src/services/school/services/permissions.js

```diff
@@ -42,7 +42,7 @@ class HandlePermissions {
 	}
 
 	async patch(id, data, params) {
-		if (!Configuration.get('TEACHER_STUDENT_VISIBILITY__IS_CONFIGURABLE')) {
+		if (this.permission === 'STUDENT_LIST' && !Configuration.get('TEACHER_STUDENT_VISIBILITY__IS_CONFIGURABLE')) {
```

Comment:
How it should work for 
HandlePermissions('student', 'LERNSTORE_VIEW'));

Reply:
Then the request should pass. That was the bug.

I tried to make it more explicit now by checking the permissions for both cases in a hook.